axondeepseg/axondeepseg
# Add optional destination folder in download_models CLI command

This PR aims to add an optional flag `-p` / `--path` and resolves some of the issued while using `download_models` CLI command to retrieve the models (**SEM** and **TEM**) stored on the OSF in a specific destination folder.

**Usage**
`download_models -p destination_folder`

**Modifications:**
1. Using CLI, user is able to download the models in a specific destination folder.
2. Fixes the bug which is caused when one tries download models in the working directory.
3. Fix bug which is encountered when user tries to download model by calling `downlaod_models` cli command in directory different from root directory


**Testing Instructions**
1. First, check` -p/--path `flag works in terminal.
                 -Go to terminal, and run `download_models -p destination_folder_path`. (**Note** path to destination folder path 
                                                                                                                                                could either be absolute or relative)
                  - SEM and TEM models would be saved under the given destination folder.
2. Second, go to the working directory and set the destination folder to the working directory. 
                 - Assume the working directory to be `axondeepseg`, run the cli command `download_models` 
                   by setting the -p  flag to be current working directory, that is, 
                    `download_models -p /Users/vasudevsharma/VScode/Projects/axondeepseg`
                 - You will note that this fails in the[ latest ](https://github.com/neuropoly/axondeepseg)master branch.
3. Third, check whether additional directories are created when download_models cli command is executed in a directory different from root directory.
                 - To test this error, follow [this](https://github.com/neuropoly/axondeepseg/issues/394#issue-742513959) comment in both the `master` branch and `vs74/download_model_cli_download_folder` branch of this PR.



 Resolves #323, Resolves #389, Resolves #394
